daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -396,28 +397,15 @@ private[lf] object Pretty {
           char('}')
       case ValueVariant(mbId, variant, value) =>
         (mbId match {
-          case None => text("")
-          case Some(id) =>
-            if (verbose)
-              prettyIdentifier(id) + char(':')
-            else
-              text("")
+          case Some(id) if verbose => prettyIdentifier(id) + char(':')
+          case _ => Doc.empty
         }) +
-          (value match {
-            case ValueUnit => text(variant)
-            case _ =>
-              text(variant) + char('(') + prettyValue(true)(value) + char(')')
-          })
+          text(variant) + char('(') + prettyValue(true)(value) + char(')')
```

Comment:
No need to make a special case for `Unit` argument:
1 - The compiler does not generate such variant 
2 - One should probably use `enum` in this case.